### PR TITLE
Multisync feature

### PR DESCRIPTION
**Pull request by BenjaminSoelberg**
This feature adds automatic synchronization between the LCD and audio. The synchronization time various between one and two seconds. This can be improved in the future but I feel this is fast enough for now. 

The basic gist is that it will at runtime slightly change the LCD refresh speed to match the audio handled "frame" synchronization.
It uses a PID loop to control the fractional part of the PLL3 used by the LCD pixel clock to either slow down or catch up.
Various audio PLL2 clocks has also been updated to exactly match each desired audio sample rate. The MCLK handling in the audio PLL is now done by the HAL which simplifies the code.

Multiple emulators have been updated to either fix missing LCD refresh function calls or updated them to take full advantage of the multisync feature.

A debug feature showing the state of the multisync can be enabled in GAME->Debug menu->Multisync debug->on/off. 
It will show either a red or a green boarder while playing a game.

Please note that this PR depends on the following other PRs which include updated to a few forked submodules:

https://github.com/sylverb/LCD-Game-Emulator/pull/1
https://github.com/sylverb/retro-go-stm32/pull/8

Also note that .gitmodules has been updated to point to my forks, so this will need to be changed back during merge.

Lastly this PR is a precursor for my Tamagotchi P1 emulator coming to a cinema near you. 

I know that this might be a lot so please reach out if you have any issues, feedback or questions.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [submodule "retro-go-stm32"]
 	path = retro-go-stm32
-	url = https://github.com/sylverb/retro-go-stm32
```
Comment (sylverb) [inline .gitmodules:3]: please set it back to sylverb once changes merged into submodule

File: .gitmodules [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+	url = git@github.com:BenjaminSoelberg/retro-go-stm32.git
 [submodule "LCD-Game-Emulator"]
 	path = LCD-Game-Emulator
-	url = https://github.com/sylverb/LCD-Game-Emulator.git
```
Comment (sylverb) [inline .gitmodules:6]: please set it back to sylverb once changes merged into submodule

**Review by sylverb — APPROVED** [review]

**Comment by Shaeto** [comment]
not sure if it is correct place for complains (no issues tab) but "imho" this MR breaks GW and MSX. i see this screen when i start any GW rom, MSX roms (for ex Aleste) works very unstable. not sure how to capture ful stack trace.

**Comment by sylverb** [comment]
@Shaeto thanks for the report, I’m not using the main branch ATM as I’m developing in the filesystem_wip branch most of the time. Moreover I’m in a foreign country with no computer until next week so I can’t check.
@BenjaminSoelberg could you check this ?

**Comment by BenjaminSoelberg** [comment]
@Shaeto thank you for reporting this, @sylverb I'll take a look at it asap!

**Comment by Shaeto** [comment]
okay i will try to switch and test filesystem branch also

my build config is: 

```bash
make -j 8 EXTFLASH_SIZE_MB=60 EXTFLASH_OFFSET=4194304 INTFLASH_BANK=2 GNW_TARGET=zelda flash
```

**Comment by BenjaminSoelberg** [comment]
@Shaeto Can you try and do a "make clean && git submodule update --recursive" prior to building and flashing ?
And just a side note, I needed to use overclocking to play Aleste as it uses more than 100% cpu.

**Comment by Shaeto** [comment]
yes i did it first time update --recursive, second time full clone --recursive and rebuild

**Comment by BenjaminSoelberg** [comment]
@Shaeto I can't reproduce it at the moment, may I suggest that you join the Discord server to ease the communication ?
We can then do it side by side which would help a lot.
https://discord.gg/XM9wKxcJ

I'm "benjamin.soelberg"

**Pull request merged**